python/ray/train/v2/tests/test_controller.py

```diff
@@ -80,8 +85,9 @@ def finish_worker(self, worker_index):
         status = self._worker_statuses[worker_index]
         status.running = False
 
-    def set_start_failure(self, start_failure):
-        self._start_failure = start_failure
+    @classmethod
+    def set_start_failure(cls, start_failure):
+        cls._start_failure = start_failure
```

Comment:
curious why this was needed?

Reply:
I needed to do this because the instance of the `WorkerGroup` is instantiated internally within the controller and changes over time. In `test_worker_group_start_failure` I had to inject this failure logic somehow so I did it by setting this and monkeypatching it.